### ISSUE TITLE
Specified cast is not valid when using QueryAsync

### ISSUE THREAD

**Issue by pranavmehta** [StackExchange/dapper-dot-net#346]
I have a stored procedure that selects a single big int - something like: select 10 as 'Id'.

I have the following code which works perfectly fine:

```C#
long Id = Conn.Query<long>("GetId", new { foo = bar }, commandType: CommandType.StoredProcedure).First();
```

However, I am unable to get it to work with QueryAsync. It always fails with "The specified cast is not valid" exception. What is the correct way to use the Async version of above code? 

Note: I am able get QueryAsync to work when using complex types, but fails for simple types like in above example

**Comment by tms** [comment]
Hmm, @mgravell, is this just a case of needing to add [the `ChangeType` fallback](https://github.com/StackExchange/dapper-dot-net/blob/master/Dapper%20NET40/SqlMapper.cs#L1642) to where the [async results are read](https://github.com/StackExchange/dapper-dot-net/blob/master/Dapper%20NET45/SqlMapperAsync.cs#L530)? I assume the DB is actually returning something that isn't directly a `long` in this case, like the situations that required the change in the synchronous code before.

**Comment by NickCraver** [comment]
@tms yep pretty close - I was making sure this wasn't already solved somewhere but it has been - in PR #338. I'll merge it in shortly.
